src/tools/rest-backend/service_config.cpp

```diff
@@ -36,12 +39,73 @@ cppcms::json::value ConfigEngine::loadApplicationConfiguration (const std::strin
 	kdb.get (ks, conf_root);
 	ks = ks.cut (kdb::Key (conf_root, KEY_END));
 
-	for (auto elem : ks)
+	cppcms::json::value result = this->transformKeysetToJsonValue (ks, conf_root);
+
+	return result;
+}
+
+/**
+ * @brief can be used to transform a kdb::KeySet into cppcms::json::value
+ * 
+ * will iterate through the keyset and use a helper method to add the key
+ * values to the json::value
+ * 
+ * @param ks the keyset that needs to be transformed
+ * @return a cppcms::json::value containing the configuration from the keyset
+ */
+cppcms::json::value ConfigEngine::transformKeysetToJsonValue (const kdb::KeySet & ks, const std::string & conf_root) const
+{
+	cppcms::json::value result;
+	kdb::KeySet ksDup = ks.dup ();
+	kdb::KeySet ksToVisit = ksDup.cut (kdb::Key ("spec/", KEY_END));
+
+	// all spec keys
+	for (auto elem : ksToVisit)
 	{
-		std::string key = elem.getName ().substr (elem.getName ().find (conf_root) + conf_root.length () + 1);
-		std::replace (key.begin (), key.end (), '/', '.');
+		kdb::Key elemCascading = elem.dup ();
+		kdb::tools::helper::removeNamespace (elemCascading);
+		kdb::Key key = ksDup.lookup (elemCascading.getName ());
```

Comment:
Can you elaborate why you chose to handle spec in a special way? I do not think this is necessary or can yield correct results.

By looking up in ksDup (which does not have spec keys anymore) the lookup cannot honor the specification anymore.